src/dex/spectrum.ts

```diff
@@ -0,0 +1,209 @@
+import { LiquidityPool } from './models/liquidity-pool';
+import { BaseDataProvider } from '@providers/data/base-data-provider';
+import { Asset, Token } from './models/asset';
+import { BaseDex } from './base-dex';
+import { DatumParameters, DefinitionConstr, DefinitionField, PayToAddress, RequestConfig, SwapFee, UTxO } from '@app/types';
+import { DefinitionBuilder } from '@app/definition-builder';
+import { AddressType, DatumParameterKey } from '@app/constants';
+import { BaseApi } from '@dex/api/base-api';
+import pool from './definitions/spectrum/pool';
+import order from './definitions/spectrum/order';
+import { all, bignumber, BigNumber, ConfigOptions, create, FormatOptions, MathJsStatic } from 'mathjs';
+
+export class Spectrum extends BaseDex {
+  public static readonly identifier: string = 'Spectrum';
+  public readonly api: BaseApi;
+
+  /**
+   * On-Chain constants.
+   */
+  public readonly orderAddress: string = 'addr1wynp362vmvr8jtc946d3a3utqgclfdl5y9d3kn849e359hsskr20n';
+  public readonly poolAddress: string = 'addr1x8nz307k3sr60gu0e47cmajssy4fmld7u493a4xztjrll0aj764lvrxdayh2ux30fl0ktuh27csgmpevdu89jlxppvrswgxsta';
```

Comment:
We'll need to add the `addr1x94ec3t25egvhqy2n265xfhq882jxhkknurfe9ny4rl9k6dj764lvrxdayh2ux30fl0ktuh27csgmpevdu89jlxppvrst84slu` address as well